src/bodywork/workflow_execution.py

```diff
@@ -301,16 +306,25 @@ def _run_batch_stages(
         _log.info(f"Creating k8s job for stage = {job_name}")
         k8s.create_job(job_object)
     try:
-        timeout = max(stage.max_completion_time for stage in batch_stages)
-        k8s.monitor_jobs_to_completion(job_objects, timeout + TIMEOUT_GRACE_SECONDS)
-    except TimeoutError as e:
+        timeout = _compute_optimal_job_timeout(batch_stages)
+        timeout_dt = (datetime.now() + timedelta(seconds=timeout)).strftime(
+            "%d/%m/%y %H:%M:%S"
+        )
+        _log.info(f"Monitoring k8s jobs, timeout at [{timeout_dt}] ({timeout}s)")
+        k8s.monitor_jobs_to_completion(job_objects, timeout)
+        for job_object in job_objects:
+            job_name = job_object.metadata.name
+            _log.info(f"Completed k8s job for stage = {job_name}")
+            _print_logs_to_stdout(namespace, job_name, False)
+    except Exception as e:
```

Comment:
Shouldn't this be `(TimeoutError, BodyworkJobFailure)` to catch everything from `monitor_jobs_to_completion()`? I'm also cool with a general Exception in this case, if there is a possibility that a legitimate unexpected Exception could be thrown by `monitor_jobs_to_completion()` with retrievable pod logs. 

Otherwise there should be a different handling for general exceptions.

Reply:
I agree with your suggestions.

`monitor_jobs_to_completion` could throw a k8s API exception, but this isn't where we should be handling it (we should do that within the function itself (job for another day).